### DISCUSSION TITLE
REQUEST: Support http://abcdefghijkl.oinion server addresses

### DISCUSSION THREAD

**Discussion by WinstonSmith73** [tateisu/SubwayTooter#180] [Ideas]
I have a self-hosted Mastodon server that runs over the Tor network, and I cannot find any Mastodon clients that will work with it because none support http .onion addresses. Please add this functionality, as the Tor protocol is inherently more secure than https. Thanks for your consideration 🙏

**Comment by tateisu** [comment]
this is android OS problem about DNS name to IP address resolution.
see also https://github.com/jhy/jsoup/issues/1174

Is there workaround for this? what is correct way to resolve DNS name of ".onion" ?

**Reply by tribela** [reply]
Tusky and Fedilab support proxy(http or socks5) for onion address

**Comment by tateisu** [comment]
I googled "android okhttp tor onion", but good solution not found.

**Comment by tateisu** [comment]
I'm not familiar with those environments so don't expect me to implement them myself

**Comment by tribela** [comment]
People also can use orbot's VPN feature to connect onion addresses from any app they want.
It does not require existing apps to be changed.

Simple solution: Use orbot's VPN feature by user self
